app/src/main/java/com/cornellappdev/transit/ui/viewmodels/HomeViewModel.kt

```diff
@@ -44,23 +45,44 @@ class HomeViewModel @Inject constructor(
      */
     val lastRouteFlow = routeRepository.lastRouteFlow
 
-    private val _queryFlow: MutableStateFlow<List<Stop>> = MutableStateFlow(emptyList())
+    /**
+     * The current query in the search bar, as a StateFlow
+     */
+    val searchQuery: MutableStateFlow<String> = MutableStateFlow("")
+
 
     /**
-     * Search query filtered flow of all TCAT stops
+     * Flow of queries
      */
-    val queryFlow = _queryFlow.asStateFlow()
+    private val _queryFlow: Flow<List<Stop>> =
+        stopFlow.combine(searchQuery) { allStops, filter ->
+            when (allStops) {
+                is ApiResponse.Error -> {
+                    emptyList()
+                }
+
+                is ApiResponse.Pending -> {
+                    emptyList()
+                }
 
+                is ApiResponse.Success -> {
+                    allStops.data.filter { stop ->
+                        fulfillsQuery(stop, filter)
+                    }
+                }
+            }
+        }
```

Comment:
Oookay yes this looks perfect! But one last thing; I typically advocate for converting the `Flow` to a `StateFlow` in the ViewModel so that the UI doesn't have to determine an initial value itself. As one last touch, please convert this to a `StateFlow` with `.stateIn(...)`:

```kotlin
.stateIn(
    scope = viewModelScope,
    started = SharingStarted.Eagerly,
    initialValue = emptyList()
)
```